lib/services/elasticsearch.js

```diff
@@ -687,6 +687,58 @@ class ElasticSearch extends Service {
     }
   }
 
+  /**
+  * Updates all documents matching the provided filters
+  *
+  * @param {String} index - Index name
+  * @param {String} collection - Collection name
+  * @param {Object} query - Query to match documents
+  * @param {Object} changes - Changes wanted on documents
+  * @param {Object} options - from (undefined), size (undefined), refresh (undefined), timeout (undefined), userId (undefined)
+  *
+  * @returns {Promise.<Object>} { documents, total, failures: [ _shardId, reason ] }
+  */
+  async updateByQuery(
+    index,
+    collection,
+    query,
+    changes,
+    { refresh, from, size } = {})
+  {
+
+    try {
+      const esRequest = {
+        index: this._getESIndex(index, collection),
+        body: this._sanitizeSearchBody({ query }),
+        scroll: '5s',
+        from,
+        size
+      };
+
+      const documents = await this._getAllDocumentsFromQuery(esRequest);
+      documents.forEach(document => {
```

Comment:
NP: You have an array as input and array as output. Why don't you use `map` function?

Reply:
Recreate a new array with map is slower and also consume a lot of memory, it's better to use `Array.forEach` to mutate the original array since this method could handle very large array (up to 10k elem)